src/client/dfs/dfs.c

```diff
@@ -831,49 +851,51 @@ open_file(dfs_t *dfs, daos_handle_t th, dfs_obj_t *parent, int flags,
 		if (rc != 0) {
 			D_ERROR("daos_array_open_with_attr() failed (%d)\n",
 				rc);
-			return daos_der2errno(rc);
+			D_GOTO(out, rc = daos_der2errno(rc));
 		}
 
 		/** Create and insert entry in parent dir object. */
 		entry->mode = file->mode;
 		entry->atime = entry->mtime = entry->ctime = time(NULL);
 		entry->chunk_size = chunk_size;
 
-		rc = insert_entry(parent->oh, th, file->name, len, entry);
+		rc = insert_entry(parent->oh, th, file->name, len,
+				  (!dfs->use_dtx || oexcl) ?
+				  DAOS_COND_DKEY_INSERT : 0, entry);
```

Comment:
What happens if we pass flags=0 (use_dtx) and the entry already exists? In this case, rc=0, so we commit and return. But will entry->mode, entry->chunk_size, etc. be in sync? I.e. the entry already exists, but with different values.

Reply:
if the entry exists, we won't be here. would be caught by the fetch.
if it was inserted by someone else after the fetch was called above, then the commit will fail with DER_RESTART, we will restart and the fetch will return exist, and we open the existing file. that is all of course if we do not have o_excl